raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -931,35 +930,53 @@ def handle_init(
     return TransitionResult(iteration.new_state, events)
 
 
-def handle_block(channelidentifiers_to_channels, state, state_change, block_number):
+def handle_block(
+        mediator_state: MediatorTransferState,
+        state_change: Block,
+        channelidentifiers_to_channels: typing.ChannelMap,
+        pseudo_random_generator: random.Random,
+        block_number: typing.BlockNumber,
+):
     """ After Raiden learns about a new block this function must be called to
     handle expiration of the hash time locks.
     Args:
         state (MediatorTransferState): The current state.
     Return:
         TransitionResult: The resulting iteration
     """
-    assert state_change.block_number == block_number
+    balance_proof = mediator_state.transfers_pair[0].payee_transfer.balance_proof
+    channel_identifier = balance_proof.channel_identifier
+    channel_state = channelidentifiers_to_channels.get(channel_identifier)
+
+    assert channel_state
```

Comment:
Hmm why an assert here? This would crash the node. Why not return an empty transition?

Reply:
```
channelidentifiers_to_channels.get(channel_identifier)
```
might return None... so the assert is just to make sure that we get a valid channel state.